doc/develop/tools/vscode.rst

```diff
@@ -0,0 +1,77 @@
+.. _vscode_ide:
+
+VS Code
+#######
+
+`Visual Studio Code`_ is a popular cross-platform IDE that supports C projects and has a rich set of
+extensions.
+
+This guide describes the process of setting up VS Code for Zephyr's
+:zephyr:code-sample:`blinky` sample in VS Code.
+
+The instructions have been tested on Linux. In terms of the VSCode workflow, the steps would be the
+same for macOS and Windows, but make sure to adjust the paths if needed.
+
+Get VS Code
+***********
+
+`Download VS Code`_ and install it.
+
+Install the required extensions through the `Extensions` marketplace in the left panel.
+Search for the `C/C++ Extension Pack`_ and install it.
+
+Initialize a new workspace
+**************************
+
+This guide gives details on how to configure the :zephyr:code-sample:`blinky`
+sample application, but the instructions would be similar for any Zephyr project and :ref:`workspace
+layout <west-workspaces>`.
+
+Before you start, make sure you have a working Zephyr development environment, as per the
+instructions in the :ref:`getting_started`.
+
+Open the project in VS Code
+***************************
+
+#. In VS Code, select :menuselection:`File --> Open Folder` from the main menu.
+
+#. Navigate to your Zephyr workspace (i.e.the :file:`zephyrproject` folder in your HOME directory if
+   you have followed the Getting Started instructions), then select
+   :file:`zephyr/samples/basic/blinky` or another sample project folder.
+
+#. If prompted, enable workspace trust.
+
+Generate compile commands
+*************************
+
+#. The setup showcased in this document relies on the `compile_commands.json` file generated by CMake
+   to provide code navigation and linting capabilities. To generate this file, you need to enable the
+   generation of this file through the following configuration:
+
+#. Open your projects' CMakeLists.txt file and add the following line.
+   In our case, we will add it to the :file:`samples/basic/blinky/CMakeLists.txt` file.
+   ``set(CMAKE_EXPORT_COMPILE_COMMANDS ON)``
+
+#. Build the project to generate the `compile_commands.json` file:
+   ``west build -b native_posix_64 samples/basic/blinky``
```

Comment:
Are you sure `compile_commands.json` is not already automatically generated? I think it's effectively always on?
https://github.com/zephyrproject-rtos/zephyr/blob/main/cmake/modules/kernel.cmake#L123-L126

```suggestion
In order to support code navigation and linting capabilities, you must compile your project once to
generate the :file:`compile_commands.json` file that will provide the C/C++ extension with the
required information (ex. include paths):

   .. zephyr-app-commands::
      :zephyr-app: samples/basic/blinky
      :board: native_sim/native/64
      :goals: build
      :compact:
```

Reply:
Yes it looks like this option has become the default. Integrating your suggestion which is using the neat zephyr-app-commands macro I'll keep in mind ;)